ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -423,6 +428,7 @@ private void removeDataStream(ClientInvocationId invocationId, StreamInfo info)
     if (info != null) {
       info.getDivision().getDataStreamMap().remove(invocationId);
       info.getLocal().cleanUp();
+      info.applyToRemotes(out -> out.out.closeAsync());
```

Comment:
Let's add a `cleanUp()` method to `StreamInfo`.
```java
    void cleanUp(ClientInvocationId invocationId) {
      getDivision().getDataStreamMap().remove(invocationId);
      getLocal().cleanUp();
      applyToRemotes(remote -> remote.out.closeAsync());
    }
```

Reply:
Updated, PTAL